anap1003/assignment3/Dictionary.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package assignment3;
+
+/**
+ *
+ * @author Ana
+ */
+public class Dictionary {
+    
+    private TrieNode root;
+    
+    public Dictionary() {
+        root = new TrieNode(null, false);
+    }
+    
+    public void addWord(String word) {
+        if (root == null) {
+            root = new TrieNode(null, false);
+        }
+        
+        TrieNode tempNode = root;
+        boolean dictionaryContainsPrefix = true;
+        word = word.toUpperCase();
+        
+        for (int i = 0; i < word.length(); i++) {
+            if (dictionaryContainsPrefix && tempNode.hasChild(word.charAt(i))) {

```

Comment:
What speed improvement does dictionaryContainsPrefix actually grant here? Adding words to the dictionary is likely a rare occurrence and the code would be simpler without it.